src/server/scripts/Pandaria/GateoftheSettingSun/instance_gate_setting_sun.cpp

```diff
@@ -0,0 +1,399 @@
+    {GO_KIPTILAK_WALLS,                     DATA_KIPTILAK,              DOOR_TYPE_ROOM, },
+    {GO_KIPTILAK_EXIT_DOOR,                 DATA_KIPTILAK,              DOOR_TYPE_PASSAGE, },
+    {GO_RIMAK_AFTER_DOOR,                   DATA_RIMOK,                 DOOR_TYPE_ROOM, },
+    {GO_RAIGONN_DOOR,                       DATA_RAIGONN,               DOOR_TYPE_ROOM, },
+    {GO_RAIGONN_AFTER_DOOR,                 DATA_RAIGONN,               DOOR_TYPE_PASSAGE, },
+    {0,                                     0,                          DOOR_TYPE_ROOM, },// END
+};
+
+class instance_gate_setting_sun : public InstanceMapScript
+{
+public:
+    instance_gate_setting_sun() : InstanceMapScript("instance_gate_setting_sun", 962) { }
+
+    InstanceScript* GetInstanceScript(InstanceMap* map) const
+    {
+        return new instance_gate_setting_sun_InstanceMapScript(map);
+    }
+
+    struct instance_gate_setting_sun_InstanceMapScript : public InstanceScript
+    {
+        uint64 kiptilakGuid;
+        uint64 gadokGuid;
+        uint64 rimokGuid;
+        uint64 raigonnGuid;
+        uint64 raigonWeakGuid;
+
+        uint64 firstDoorGuid;
+        uint64 fireSignalGuid;
+
+        uint64 wallCGuid;
+        uint64 portalTempGadokGuid;
+
+        uint32 cinematicTimer;
+        uint8 cinematicEventProgress;
+
+        std::list<uint64> bombarderGuids;
+        std::list<uint64> bombStalkerGuids;
+        std::list<uint64> mantidBombsGUIDs;
+        std::list<uint64> rimokAddGenetarorsGUIDs;
+        std::list<uint64> artilleryGUIDs;
+        std::list<uint64> secondaryDoorGUIDs;
+
+        uint32 dataStorage[MAX_DATA];
+
+        instance_gate_setting_sun_InstanceMapScript(Map* map) : InstanceScript(map)
+        {}
+
+        void Initialize()
+        {
+            SetBossNumber(EncounterCount);
+            LoadDoorData(doorData);
+
+            kiptilakGuid            = 0;
+            gadokGuid               = 0;
+            rimokGuid               = 0;
+            raigonnGuid             = 0;
+            raigonWeakGuid          = 0;
+            
+            firstDoorGuid           = 0;
+
+            cinematicTimer          = 0;
+            cinematicEventProgress  = 0;
+
+            wallCGuid               = 0;
+            portalTempGadokGuid     = 0;
+
+            memset(dataStorage, 0, MAX_DATA * sizeof(uint32));
+
+            bombarderGuids.clear();
+            bombStalkerGuids.clear();
+            mantidBombsGUIDs.clear();
+            rimokAddGenetarorsGUIDs.clear();
+            artilleryGUIDs.clear();
+            secondaryDoorGUIDs.clear();
+        }
+
+        void OnDestroy(InstanceMap* pMap)
+        {
+            if (Creature* weakSpot = instance->GetCreature(GetObjectGuid(NPC_WEAK_SPOT)))
+                weakSpot->_ExitVehicle();
+        }
+
+        void OnPlayerEnter(Player* player)
+        {
+            if (GetData(DATA_BRASIER_CLICKED) == NOT_STARTED)
+                player->SetPhaseMask(1, true);

```

Comment:
there is no phaseMask on 6x, please don't use it, there are just ancient methods in core